packages/calcite-components/src/components/block-group/block-group.tsx

```diff
@@ -319,7 +328,23 @@ export class BlockGroup extends LitElement implements InteractiveComponent, Sort
   }
 
   private handleDefaultSlotChange(event: Event): void {
-    updateBlockChildren(event.target as HTMLSlotElement);
+    this.blockAndGroups = slotChangeGetAssignedElements(event).filter(
+      (el): el is Block["el"] | BlockGroup["el"] =>
+        el.matches(blockSelector) || el.matches(blockGroupSelector),
+    );
+
+    const blockChildren = this.blockAndGroups.filter((block): block is Block["el"] =>
```

Comment:
You already filter block and block groups above, so you could build `blockChildren` in that earlier filter pass. This makes my suggestion above moot.